bpf/process/types/basic.h

```diff
@@ -521,6 +521,8 @@ __copy_char_buf(long off, unsigned long arg, unsigned long bytes,
 
 	/* Bound bytes <4095 to ensure bytes does not read past end of buffer */
 	rd_bytes = bytes;
+	while (rd_bytes > 0xfff)
+		rd_bytes >>= 1;
```

Comment:
nice catch.. but that will break 4.19 kernels that don't have loops, could we just do:

       if (rd_bytes > 0xfff)
               rd_bytes = 0xfff;

Reply:
i have tried this way,  but resulted in the error 
```
error="failed prog /var/lib/tetragon/bpf_generic_kprobe_v53.o kern_version 328823 LoadKprobeProgram: opening collection '/var/lib/tetragon/bpf_generic_kprobe_v53.o' failed: program generic_kprobe_process_event4: load program: permission denied: 1241: (85) call bpf_probe_read#4: R2 min value is negative, either use unsigned or 'var &= const' (2439 line(s) omitted)"
```
BPF Verifier not paas. 
5.4 kernels.

Reply:
https://git.kernel.org/pub/scm/linux/kernel/git/netdev/net-next.git/commit/kernel/bpf/verifier.c?id=06c1c049721a995dee2829ad13b24aaf5d7c5cce

Reply:
Although I use unsigned set 
``` 
unsigned len = 0xfff;
err = probe_read(&s[2], len, (char *)arg);
```
, but still can not pass the BPF verifier .
It seems that I can only operate on the ```bytes``` already in the register, but not allowed to modify its value directly.

Reply:
i change  to use recursion instead of loops. 
after testing, the result is the same as using loops.

Reply:
@olsajiri beautiful, it solves the problem even better.

Reply:
@Y-dc do you want to submit that change (preferred) or should I follow up on that?

Reply:
@olsajiri thanks, i submitted